src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -199,12 +196,6 @@ def login_with_service_principal(self, client_id, credential, scopes):
         entry = sp_auth.get_entry_to_persist()
         self._service_principal_store.save_entry(entry)
 
-    def login_with_managed_identity(self, scopes, identity_id=None):  # pylint: disable=too-many-statements
-        raise NotImplementedError
-
-    def login_in_cloud_shell(self, scopes):
-        raise NotImplementedError
```

Comment:
Why do we need to remove `login_in_cloud_shell` Since cloud_shell is a special user identity in msal?

Reply:
I don't agree with MSAL's design, see https://github.com/Azure/azure-cli/pull/29637#issuecomment-2350937911, https://github.com/Azure/azure-cli/pull/29637#discussion_r1998291449 for detailed explanations.

I want to keep the calling pattern for `Identity` consistent - `authority` should always be a required parameter.

For Cloud Shell authentication, the CLI command is `az login --identity`. It **is** a managed identity credential, no matter what token is returned. Including `CloudShellCredential` in `Identity` blurs its boundary.